docs/proposals/cross-mesh-ux.md

```diff
@@ -0,0 +1,241 @@
+          - mesh:                    # and/or a new entry type here
+            - name: consume
+              tags:
+                kuma.io/service: client
+        backends:
+          - destination:
+              kuma.io/service: server
+      - matches:    # ALL matches must succeed
+          - path:
+              match: PREFIX
+              value: /api/v2
+          - mesh:
+            - name: *
+        backends:
+          - destination:
+              kuma.io/service: server
+```
+
+The advantage of this option is probably reduced complexity for less complicated
+cases.
+
+### `consume` UX
+
+In order for `consume` services to communicate with `expose`, we need to give
+those services a way to contact the `MeshGateway`. A `MeshGateway` doesn't have
+an address or concrete identity beyond the identities of the `kuma-dp` proxies that
+serve the gateway.
+
+#### Options
+
+There are multiple UX options for making the `MeshGateway` addressable from
+other meshes.
+
+##### Virtual outbounds
+
+We can add support for `MeshGateway` listeners to `VirtualOutbounds` to give applications
+a way to address a `MeshGateway` listener from another mesh.
+
+The `VirtualOutbound` selects `Dataplane`s and uses them to generate outbounds.
+These generated outbounds are added to _all_ `Dataplane` resources.
+
+```
+---
+type: VirtualOutbound
+mesh: consume
+name: test
+selectors: # this VirtualOutbound will be built using the matched data plane proxies and will be added for all data plane proxies
+  - match:
+      kuma.io/service: edge-gateway
+      name: mesh-listener
+conf:
+  host: "gateway.{{.gateway.mesh}}.mesh"
+    # or host: "{{.gateway.listener.hostname}}"
+  port: "{{.gateway.listener.port}}"
+  parameters:
+    - name: service
+      tagKey: "kuma.io/service"
+    - name: gateway
+      gatewayListener: {}
+```
+
+At the moment, the `selectors` are only run for `Dataplane` resources whereas we
+need support for `crossMesh`-enabled `MeshGateway` listeners (of all `Mesh`es) as well. In the
+implementation, this means keeping track of all `MeshGateway`-serving `Dataplane`s
+of _any mesh_ when generating a `VirtualOutbound` of _any mesh_.
+
+We also need to provide the properties of the `MeshGateway` listener to the `host`
+template in order to generate the hostname.
+
+This should give us a virtual outbound:
+
+```
+address: 240.0.0.1
+port: 8090
+mesh: expose           # we need to keep track of the mesh somehow
+tags:
+  kuma.io/service: edge-gateway
+  name: mesh-listener
+```
+
+along with the hostname `gateway.expose.mesh`, giving access from `consume` to `expose`
+at `http://gateway.expose.mesh:8090/api`.
+
+##### Listener hostname
+
+Another option for giving access to the `MeshGateway` would be to require and
+use the `listeners[].hostname` option to generate an outbound for every 
+dataplane implicitly. In effect, this would be a virtual outbound with the
+`host` template equal to `{{ .gateway.listener.hostname }}`.
```

Comment:
I really like this idea. Honestly? For the initial implementation, I think I'd go this way and drop explicit VirtualOutbound. This gives users flexibility (they choose their own hostname after all) and it is very explicit